src/AnnotatedCommand.php

```diff
@@ -225,8 +225,9 @@ protected function setCommandArguments($commandInfo)
      */
     protected function checkUsesInputInterface($params)
     {
+        /** @var \ReflectionParameter $firstParam */
         $firstParam = reset($params);
-        return $firstParam instanceof InputInterface;
+        return $firstParam->getClass() && $firstParam->getClass()->implementsInterface(InputInterface::class);
```

Comment:
You are correct, `reset($params)` returns `false` or the first parameter. This is taken into account in the check now.